lib/Utils/SourceNormalization.cpp

```diff
@@ -375,12 +375,32 @@ size_t cling::utils::getWrapPoint(std::string& source,
       return std::string::npos;
     }
 
-    const tok::TokenKind kind = Tok.getKind();
     // Prior behavior was to return getFileOffset, which was only used as an
     // in a test against std::string::npos. By returning 0 we preserve prior
     // behavior to pass the test against std::string::npos and wrap everything
     const size_t offset = 0;
 
+    // Check, if a function with c++ attributes should be defined.
+    while (Tok.getKind() == tok::l_square) {
```

Comment:
My code should not affect the structured binding construct, because it only works, if the statement starts with `[`.

The idea of my implementation is, that I can safety detect the `[` of the c++ attribute at the begin of a function definition. If the syntax of the attributes are okay, the loop iterate over the tokens, until to the first token, which doesn't match the c++ attribute syntax. For example before my loop the string to lex is `[[ deprecated ]] [[ noreturn ]] inline void foo() {}` and after the loop is it `inline void foo() {}`.

Reply:
@Axel-Naumann if the token are defined by macros, how would we get them in here?